libs/layout/src/lib/layout.tsx

```diff
@@ -1,27 +1,19 @@
 import Footer from './footer/footer';
-import { PropsWithChildren, useCallback, useState } from 'react';
+import { PropsWithChildren } from 'react';
 import { Outlet } from 'react-router';
 import Sidebar from './sidebar/sidebar';
 import Header from './header/header';
-import { useAuth } from '@pasnik/auth';
-import { FullscreenSpinner } from '@pasnik/components';
+import { useLayoutStore } from './layout.store';
 
 export interface LayoutProps {
   version?: string;
 }
 
 export function Layout({ version }: PropsWithChildren<LayoutProps>) {
-  const [sidebarOpen, setSidebarOpen] = useState(false);
-  const { user } = useAuth();
-  const closeSidebar = useCallback(() => {
-    setSidebarOpen(false);
-  }, []);
-  const openSidebar = useCallback(() => {
-    setSidebarOpen(true);
-  }, []);
-  return !user ? (
-    <FullscreenSpinner />
-  ) : (
+  const sidebarOpen = useLayoutStore((store) => store.sidebarOpen);
```

Comment:
second parameter of zustand selector is the equality fn could use that to memoize the result

Reply:
via docs

It detects changes with strict-equality (old === new) by default, this is efficient for atomic state picks.